pkg/graveler/ref/manager.go

```diff
@@ -3,36 +3,59 @@ package ref
 import (
 	"context"
 	"errors"
+	"fmt"
 	"time"
 
 	"github.com/treeverse/lakefs/pkg/db"
 	"github.com/treeverse/lakefs/pkg/graveler"
 	"github.com/treeverse/lakefs/pkg/ident"
+	"golang.org/x/sync/singleflight"
 )
 
 // IteratorPrefetchSize is the amount of records to maybeFetch from PG
 const IteratorPrefetchSize = 1000
 
+// MaxBatchDelay wait time before perform database read operation, it enables batching requests at that period of time
+const MaxBatchDelay = 3 * time.Millisecond
```

Comment:
Consider
```suggestion
const BatchDelay = 3 * time.Millisecond
```
The "max" is a bit confusing.